src/api/Methods.ts

```diff
@@ -56,8 +57,16 @@ export const fdc3Ready = async (waitForMs = DEFAULT_TIMEOUT): Promise<void> => {
   });
 };
 
-export function open(app: TargetApp, context?: Context): Promise<void> {
-  return rejectIfNoGlobal(() => window.fdc3.open(app, context));
+function isString(app: AppIdentifier | String): app is String {
+  return typeof app === 'string';
+}
+
+export function open(app: AppIdentifier | String, context?: Context): Promise<AppMetadata> {
+  if (isString(app)) {
+    return rejectIfNoGlobal(() => window.fdc3.open(app, context));
```

Comment:
Shouldn't these overloads be wrapping the string as an AppIdentifier internally?  Otherwise why is there a check and if-else?

Reply:
The typeguard and check were needed as typescript will allow us to declare the different types for the function overload, but in the implementation, javascript won't (the second declaration will just overwrite the first, which typescript knows and will spit out as an error).  

Hence, this looks odd in this utility file, but works correctly and looks correct in the types/docs (and provides the overload for use in other languages that actually care about the types of function arguments, which JS does not).  Hence, JS implementations  will still use a single function to implement both signatures and will have to do some similar type checking internally.

Makes a bit more sense when you see it in VS code and it highlights that one of the calls fdc3.open is the deprecated form:
![image](https://user-images.githubusercontent.com/1701764/169286340-a82239eb-f74b-482e-9382-076c7ad66ce2.png)

I can't wrap the `name` as an `AppIdentifier` internally as we're missing the required `appId`.